lib/model/model_test.go

```diff
@@ -2035,13 +2003,64 @@ func TestGlobalDirectorySelfFixing(t *testing.T) {
 
 	m.Index(device1, "default", testdata)
 
-	result := m.GlobalDirectoryTree("default", "", -1, false)
-
-	if mm(result) != mm(expectedResult) {
-		t.Errorf("Does not match:\n%s\n%s", mm(result), mm(expectedResult))
-	}
-
-	result = m.GlobalDirectoryTree("default", "xthis/is/a/deep", -1, false)
+	actual := m.GlobalDirectoryTree("default", "", -1, false)
+	expected := []*DirectoryTree{
+		mkdir("another", []*DirectoryTree{
+			mkdir("directory", []*DirectoryTree{
+				mkfile("afile"),
+				mkdir("with", []*DirectoryTree{
+					mkdir("a", []*DirectoryTree{
+						mkfile("file"),
+					}),
+					mkfile("file"),
+				}),
+			}),
+		}),
+		mkdir("other", []*DirectoryTree{
+			mkdir("random", []*DirectoryTree{
+				mkdir("dirx", nil),
+			}),
+			mkdir("randomx", nil),
+		}),
+		mkdir("some", []*DirectoryTree{
+			mkdir("directory", []*DirectoryTree{
+				mkdir("with", []*DirectoryTree{
+					mkdir("a", []*DirectoryTree{
+						mkfile("file"),
+					}),
+					mkdir("x", nil),
+				}),
+			}),
+		}),
+		mkdir("this", []*DirectoryTree{
+			mkdir("is", []*DirectoryTree{
+				mkdir("a", []*DirectoryTree{
+					mkdir("deep", []*DirectoryTree{
+						mkdir("invalid", []*DirectoryTree{
+							mkdir("directory", nil),
+						}),
+					}),
+				}),
+			}),
+		}),
+		mkdir("xthis", []*DirectoryTree{
+			mkdir("is", []*DirectoryTree{
+				mkdir("a", []*DirectoryTree{
+					mkdir("deep", []*DirectoryTree{
+						mkdir("invalid", []*DirectoryTree{
+							mkfile("file"),
+						}),
+					}),
+				}),
+			}),
+		}),
+	}
+
+	if mm(actual) != mm(expected) {
+		t.Errorf("Does not match:\n%s\n%s", mm(actual), mm(expected))
+	}
```

Comment:
I suspect this should stop working, as I think your current implementation breaks this "feature"

Reply:
Yes, this test function (`TestGlobalDirectorySelfFixing`) is failing. wasn't sure about the implementation in those test cases, therefore I rather asked first.